### PR TITLE
runtime: Fix !x86 static checks

### PR DESCRIPTION
**Pull request by Jakob-Naucke**
Follow-up of #2237 for s390x -- field alignment isn't always minimal

Depends-on: github.com/kata-containers/tests#4047
Fixes: #2773
Signed-off-by: Jakob Naucke <jakob.naucke@ibm.com>

### CODE REVIEW

**Comment by Jakob-Naucke** [comment]
let's see /test-s390x

**Comment by Jakob-Naucke** [comment]
dependency PR was invalid -- /test-s390x

**Comment by Jakob-Naucke** [comment]
okay, https://github.com/kata-containers/tests/pull/4048 is in. /test-s390x

**Review by cmaf — DISMISSED** [review]

**Comment by cmaf** [comment]
@Jakob-Naucke this is in the s390x output: 
```
[static-checks.sh:360] INFO: Running golangci-lint on /home/jenkins/workspace/kata-containers-2.0-ubuntu-s390x-PR/go/src/github.com/kata-containers/kata-containers/src/runtime/virtcontainers
++ cd /home/jenkins/workspace/kata-containers-2.0-ubuntu-s390x-PR/go/src/github.com/kata-containers/kata-containers/src/runtime/virtcontainers
++ GO111MODULE=auto
++ eval golangci-lint 'run -c /home/jenkins/workspace/kata-containers-2.0-ubuntu-s390x-PR/go/src/github.com/kata-containers/tests/.ci/.golangci.yml' .
+++ golangci-lint run -c /home/jenkins/workspace/kata-containers-2.0-ubuntu-s390x-PR/go/src/github.com/kata-containers/tests/.ci/.golangci.yml .
qemu_arch_base.go:186:2: `protection` is unused (structcheck)
	protection           guestProtection
	^
```

**Comment by Jakob-Naucke** [comment]
Adding a `nolint` for that one -- I cannot reproduce this on x86 for whatever reason. Invalidating review due to a new commit. /test-s390x

**Comment by Jakob-Naucke** [comment]
needed a `gofmt`, /test-s390x

**Comment by Amulyam24** [comment]
Hi @Jakob-Naucke , I have one last fix for static checks on ppc64le, would you mind if I add it in this PR?(since this aligns to !x86 fixes)

**Comment by Jakob-Naucke** [comment]
@Amulyam24 fine with me. Please do add a comment in #2773 about the failure that your amendment will fix.

**Comment by Amulyam24** [comment]
Thanks @Jakob-Naucke :)

**Comment by Jakob-Naucke** [comment]
ARM failure is tracked in #2789. FC failure is tracked in #3998. /retest-fc

**Comment by Jakob-Naucke** [comment]
looks like one-off /retest-fc

**Comment by jongwu** [comment]
Hello @Jakob-Naucke -, how about picking up my commits [here](https://github.com/kata-containers/kata-containers/pull/2790/commits) in this PR to speed up review.

**Comment by Jakob-Naucke** [comment]
you get a merge, you get a merge, everyone gets a merge!!! /test

**Comment by Jakob-Naucke** [comment]
closing and reopening due to GHA not running

**Review by fidencio — CHANGES_REQUESTED** [review]
I've made the same comments I made on #2790.

Once those are addressed we can have this one merged.
Thanks both @Jakob-Naucke and @jongwu!

**Comment by jongwu** [comment]
thanks @fidencio. 
hello @Jakob-Naucke -, can you help resolve these change requests, as the [PR](https://github.com/kata-containers/kata-containers/pull/2790) is closed. I think it's easy for you.

**Comment by jongwu** [comment]
Hello @Jakob-Naucke -, I have pushed the changes to my check branch [here](https://github.com/jongwu/kata-containers/tree/check). You can pick it up, thanks!

**Comment by Jakob-Naucke** [comment]
I took the freedom to apply @fidencio's suggestion. /test

File: src/runtime/virtcontainers/qemu_arch_base.go [hunk 1]
```diff
@@ -158,19 +158,23 @@ const (
 
 	//Intel Trust Domain Extensions
 	//https://software.intel.com/content/www/us/en/develop/articles/intel-trust-domain-extensions.html
-	tdxProtection
+	// Exclude from lint checking for it won't be used on arm64 code
+	tdxProtection //nolint
```
[the remark this answers is not in the document]
Comment (Jakob-Naucke) [reply src/runtime/virtcontainers/qemu_arch_base.go:162]: more funnily, it also did not fail on s390x (and ppc64le apparently), see [my comment above](https://github.com/kata-containers/kata-containers/pull/2774#issuecomment-936092091)

[the remark this answers is not in the document]
Comment (amshinde) [reply src/runtime/virtcontainers/qemu_arch_base.go:162]: @Jakob-Naucke thanks for the confirmation. Yes, I am not sure what the lint check is failing on arm at this point.
I am not sold out on the nolint solution, however if this is blocking a lot of PRs, I am fine with opening an issue and addressing this properly in a future PR.

[the remark this answers is not in the document]
Comment (Jakob-Naucke) [reply src/runtime/virtcontainers/qemu_arch_base.go:162]: @amshinde okay, let's do it that way -- #2801

**Review by amshinde — APPROVED** [review]

**Pull request merged**